word_cloud_generator/word_cloud_generator.py

```diff
@@ -0,0 +1,254 @@
+from argparse import ArgumentParser, RawDescriptionHelpFormatter
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from pathlib import Path
+from sys import argv
+from time import sleep
+from os import path
+from wordcloud import WordCloud
+
+import json
+import requests
+import re
+
+
+def parse_args(inp_args):
+    """
+    Function to parse commandline args
+    :param inp_args: a set of commandline args to parse (dict)
+    :returns: A dictionary of parsed args
+    """
+    # Get arguments passed to the script
+    parser = ArgumentParser(formatter_class=RawDescriptionHelpFormatter)
+
+    parser.add_argument(
+        "-u",
+        "--username",
+        metavar="USERNAME",
+        help="FedID of the user",
+        required=True,
+    )
+    parser.add_argument(
+        "-p",
+        "--password",
+        metavar="PASSWORD",
+        help="Password of the user",
+        required=True,
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        metavar="OUTPUT",
+        help="Directory to create the output files in",
+        default="output",
+    )
+    parser.add_argument(
+        "-s",
+        "--start_date",
+        metavar="START_DATE",
+        help="Date to get issues from",
+        default=datetime.now().strftime("%Y-%m-%d"),
+    )
+    parser.add_argument(
+        "-e",
+        "--end_date",
+        metavar="END_DATE",
+        help="Date to get issues to",
+        default=(datetime.now() - relativedelta(months=1)).strftime("%Y-%m-%d"),
+    )
+    parser.add_argument(
+        "-a",
+        "--assigned",
+        metavar="ASSIGNED",
+        help="Assigned user to get tickets from",
+    )
+    parser.add_argument(
+        "-f",
+        "--filter_for",
+        metavar="FILTER_FOR",
+        help="Strings to filter the word cloud for",
+    )
+    parser.add_argument(
+        "-n",
+        "--filter_not",
+        metavar="FILTER_NOT",
+        help="Strings to filter the word cloud to not have",
+    )
+    args = parser.parse_args(inp_args)
+    return args
+
+
+def get_response_json(auth, headers, url):
+    """
+    Function to send a get request to a url and return the response as json
+    :param auth: A HTTPBasicAuth object for authentication (HTTPBasicAuth)
+    :param headers: A request Header (dict)
+    :param url: The URL to send the request (string)
+    :returns: A dictionary of JSON values
+    """
+    session = requests.session()
+    session.headers = headers
+    session.auth = auth
+
+    attempts = 5
+
```

Comment:
```suggestion
    response = None
```
response is referenced before assignment